rule/unused_receiver.go

```diff
@@ -21,26 +21,26 @@ type UnusedReceiverRule struct {
 func (r *UnusedReceiverRule) configure(args lint.Arguments) {
 	// while by default args is an array, i think it's good to provide structures inside it by default, not arrays or primitives
 	// it's more compatible to JSON nature of configurations
-	var allowedRegexStr string
+	var allowRegexStr string
 	if len(args) == 0 {
-		allowedRegexStr = "^_$"
+		allowRegexStr = "^_$"
 		r.failureMsg = "method receiver '%s' is not referenced in method's body, consider removing or renaming it as _"
 	} else {
 		// Arguments = [{}]
 		options := args[0].(map[string]any)
-		// Arguments = [{allowedRegex="^_"}]
+		// Arguments = [{allowRegex="^_"}]
 
-		if allowedRegexParam, ok := options["allowRegex"]; ok {
-			allowedRegexStr, ok = allowedRegexParam.(string)
+		if allowRegexParam, ok := options["allowRegex"]; ok {
+			allowRegexStr, ok = allowRegexParam.(string)
 			if !ok {
-				panic(fmt.Errorf("error configuring [unused-receiver] rule: allowedRegex is not string but [%T]", allowedRegexParam))
+				panic(fmt.Errorf("error configuring [unused-receiver] rule: allowRegex is not string but [%T]", allowRegexParam))
```

Comment:
It's not that I'm really against of renaming of all the variables here, but it seems like this line was the only one that really addressed what this PR describes in its title.

Reply:
I thought incorrect variable names had the same mistaken nature as the incorrect panic message. That's why I renamed the variables as well. I can extend the PR's title, but it will exceed the title's limit. I could create a separate PR to rename the variables, but this renaming is closely tied to fixing the panic message, so there is no need to split them.

PR titles appear in commits, which will appear in release notes. What really matters for any revive users from this commit is "fixing the panic message". Renaming local variables won't exceed the current project's scope.

Reply:
I'm down for renaming them in the same PR, as long they are in different commits, and are merged without squashing